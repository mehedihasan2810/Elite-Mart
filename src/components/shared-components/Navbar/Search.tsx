import React from 'react'
import styles from '@/scss/shared-component-styles/Navbar/Search.module.scss';

const Search = () => {
  return (
    <form className={styles.search_form} data-search-input>
    <input type="search" placeholder="Search" />
  </form>
  )
}

export default Search