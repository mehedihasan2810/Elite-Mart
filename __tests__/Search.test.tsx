import Search from "@/components/shared-components/Navbar/Search";
import { render, screen } from "@testing-library/react";

describe("<Search/>", () => {
  test("should Search component render Properly with a search field and placeholder text should be Search", () => {
    // ARRANGE
    const { asFragment } = render(<Search />);

    // ACT
    const searchField = screen.getByPlaceholderText("Search products...");

    // ASSERT
    expect(searchField).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
