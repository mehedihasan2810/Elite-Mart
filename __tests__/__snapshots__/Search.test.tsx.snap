// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search/> should Search component render Properly with a search field and placeholder text should be Search 1`] = `
<DocumentFragment>
  <form
    class="search_form"
    data-search-input="true"
  >
    <input
      placeholder="Search products..."
      type="search"
    />
  </form>
</DocumentFragment>
`;
